src/product/product.service.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { validate as isUUID } from 'uuid';
+
+import { CreateProductDto } from './dto';
+import { Product } from './entities/product.entity';
+import { PaginationDto } from '../common/dto/pagination.dto';
+
+@Injectable()
+export class ProductService {
+  constructor(
+    @InjectRepository(Product)
+    private readonly productRepository: Repository<Product>,
+  ) {}
+
+  async findOne(searchTerm: string) {
+    let product: Product;
+
+    if (isUUID(searchTerm)) {
+      product = await this.productRepository.findOneBy({ id: searchTerm });
+    } else {
+      product = await this.productRepository.findOneBy({ name: searchTerm });
+    }
+
+    if (!product) throw new BadRequestException('Product not found');
+
+    return product;
+  }
+
+  async findAll(paginationDto: PaginationDto) {
+    const { limit = 10, offset = 0, sort = 'DESC' } = paginationDto;
+
+    return await this.productRepository.find({
```

Comment:
We should never do a "return await", ill explain it on slack as well.

Reply:
Understood. I've implemented the change we discussed on slack